### PR TITLE
chore: re-organize and add tasks

### PR DESCRIPTION
**Pull request by victorolinasc**
Re-organize in contexts: brcode, payments and jws.

Also, added two mix tasks for testing from the mix CLI: `mix pix.load_dynamic` and `mix pix.read_brcode` for easing the testing of QRCodes.

### CODE REVIEW

File: lib/ex_pix_brcode/changesets.ex [hunk 1]
```diff
@@ -57,4 +58,25 @@ defmodule ExPixBRCode.Changesets do
...
 
   def valid_document?(_document, _field), do: false
+
+  @doc """
+  'chave' and 'infoAdicional' validation.
+
+  These fields have the following rules:
+    - 
+  """
+  def validate_chave_and_info_adicional_length(%{valid?: false} = c, _, _, _), do: c
```
Comment (basmoura) [inline lib/ex_pix_brcode/changesets.ex:68]: oq acha de mudar tudo para ingles e no doc especifica que por exemplo, key é o campo chave e add_info é o campo infoAdicional?
```suggestion
  def validate_key_and_additional_info_length(%{valid?: false} = c, _, _, _), do: c
```

File: lib/ex_pix_brcode/changesets.ex [hunk 2]
```diff
@@ -57,4 +58,25 @@ defmodule ExPixBRCode.Changesets do
...
+
+  def validate_chave_and_info_adicional_length(changeset, chave, info_adicional, message) do
```
Comment (basmoura) [inline lib/ex_pix_brcode/changesets.ex:70]: o mesmo aqui sobre o nome da funcao

**Review by basmoura — APPROVED** [review]

File: lib/ex_pix_brcode/changesets.ex (hunk 1 above)
Comment (victorolinasc) [reply lib/ex_pix_brcode/changesets.ex:68]: Fiquei em dúvida aqui por ser uma mistura de inglês e português...

O campo chama realmente `info_adicional` e chave. Daí pensei que indica melhor ONDE é usado deixando assim.

Comment (basmoura) [reply lib/ex_pix_brcode/changesets.ex:68]: Entendi. Eu ainda acho confuso essa mistura e como tem a documentação da função, nela poderia testar contida a referencia para o campo `chave` e para o campo `info_adicional`, mas enfim, é algo mais 💅🏽 mesmo

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            - v1-build-cache
+      - run: MIX_ENV=test mix deps.get
+      - save_cache:
+          key: v1-dependency-cache-{{ checksum "mix.lock" }}
+          paths: deps
+      - run: MIX_ENV=test mix quality.ci
+      - save_cache:
+          key: v1-build-cache-{{ .Branch }}-{{ .Environment.CIRCLE_SHA1 }}
+          paths: _build
+      - store_test_results:
```
Comment (polvalente) [inline .circleci/config.yml:28]: It's better to create caches scoped by the elixir/OTP version, which could cause some problems later on

File: lib/ex_pix_brcode/brcode/models/brcode.ex [hunk 4]
```diff
@@ -75,11 +72,13 @@ defmodule ExPixBRCode.Models.BRCode do
...
-    |> validate_inclusion(:payload_format_indicator, ["01"])
-    |> validate_inclusion(:point_of_initiation_method, ["12"])
+    |> validate_inclusion(:payload_format_indicator, ~w(01))
+    |> validate_inclusion(:point_of_initiation_method, ~w(11 12))
     |> validate_format(:merchant_category_code, ~r/^[0-9]{4}$/)
-    |> validate_inclusion(:transaction_currency, ["986"])
-    |> validate_inclusion(:country_code, ["BR"])
+    |> validate_inclusion(:transaction_currency, ~w(986))
+    |> validate_length(:transaction_amount, max: 13)
+    |> validate_format(:transaction_amount, ~r/(^0$)|(^[0-9]+\.[0-9]*$)/)
```
Comment (polvalente) [inline lib/ex_pix_brcode/brcode/models/brcode.ex:80]:
```suggestion
    |> validate_format(:transaction_amount, ~r/(^[0-9]+(\.[0-9]*)?$)/)
```

Your regex does not allow "00" or "01". Is this intentional? If not, My suggestion allows for these cases as well as "00." and "00.0"

File: lib/ex_pix_brcode/brcode/models/brcode.ex [hunk 5]
```diff
@@ -116,7 +115,7 @@ defmodule ExPixBRCode.Models.BRCode do
     model
     |> cast(params, [:reference_label])
     |> validate_required([:reference_label])
-    |> validate_length(:reference_label, min: 1, max: 25)
+    |> validate_format(:reference_label, ~r/(^[a-zA-Z0-9]{1,25}+$)|(^\*\*\*$)/)
```
Comment (polvalente) [inline lib/ex_pix_brcode/brcode/models/brcode.ex:118]:
```suggestion
    |> validate_format(:reference_label, ~r/(^[a-zA-Z0-9]{1,25}$)|(^\*\*\*$)/)
```

File: lib/ex_pix_brcode/brcode/models/brcode.ex (hunk 4 above)
Comment (victorolinasc) [reply lib/ex_pix_brcode/brcode/models/brcode.ex:80]: It is intentional. According to the specification of BACEN and the EMV-QRCPS we must allow:

- 0
- 01.
- 100.00

File: lib/ex_pix_brcode/brcode/models/brcode.ex (hunk 5 above)
Comment (polvalente) [reply lib/ex_pix_brcode/brcode/models/brcode.ex:118]: I don't think the `+` does anything here

File: lib/ex_pix_brcode/brcode/models/brcode.ex (hunk 4 above)
Comment (polvalente) [reply lib/ex_pix_brcode/brcode/models/brcode.ex:80]: Is "01" allowed by the spec?

Comment (polvalente) [reply lib/ex_pix_brcode/brcode/models/brcode.ex:80]: Maybe there should be a link to the spec as a comment in the code?

Comment (victorolinasc) [reply lib/ex_pix_brcode/brcode/models/brcode.ex:80]: The link is, unfortunatelly, very dynamic. I can "mention it" for sure.

Only "0" has special semantics. All other must have one ".".

File: .circleci/config.yml (hunk 3 above)
Comment (victorolinasc) [reply .circleci/config.yml:28]: I will leave it like that for now as it is how we use the cache in other internal libs. Certainly can be upgraded but I'd like first to have it like all others and then later we standardize it across our libraries.

**Pull request merged**